### PR TITLE
Remove org.modeshape.jcr hacks

### PR DESCRIPTION
**Pull request by acoburn**
See: https://jira.duraspace.org/browse/FCREPO-1535

### CODE REVIEW

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetBinaryStore.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * Get the internal Modeshape BinaryStore from a Repository
+ *
+ * @author acoburn
+ */
+public class GetBinaryStore implements Function<Repository, BinaryStore> {
+
+    private static final Logger LOGGER =
+            getLogger(GetBinaryStore.class);
```
Comment (ajs6f) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetBinaryStore.java]: Why is this line broken?

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetBinaryStore.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    public BinaryStore apply(final Repository repo) {
+        requireNonNull(repo, "null cannot have a BinaryStore");
+        try {
+            if (repo instanceof JcrRepository) {
+                return ((JcrRepository)repo).getConfiguration()
+                    .getBinaryStorage()
+                    .getBinaryStore();
+            }
+        } catch (final Exception ex) {
```
Comment (ajs6f) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetBinaryStore.java]: Is this really likely to be an arbitrary `Exception`?

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetCacheManager.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Get the internal Infinispan CacheManager from a Repository
+ *
+ * @author acoburn
+ */
+public class GetCacheManager implements Function<Repository, CacheManager> {
+
+    private static final Logger LOGGER =
+            getLogger(GetCacheManager.class);
```
Comment (ajs6f) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetCacheManager.java]: Why a broken line?

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetCacheManager.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    @Override
+    public CacheManager apply(final Repository repo) {
+        requireNonNull(repo, "null cannot have a CacheManager");
+        try {
+            if (repo instanceof JcrRepository) {
+                return new DefaultCacheManager(((JcrRepository)repo)
+                        .getConfiguration()
+                        .getCacheConfiguration());
+            }
+        } catch (final Exception ex) {
```
Comment (ajs6f) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetCacheManager.java]: Again, why catch `Exception`?

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/utils/FedoraTypesUtils.java [hunk 5]
```diff
@@ -122,6 +123,36 @@
...
+    public static Predicate<Node> isExternalNode = uncheck(n ->  {
+        if (NodeKey.isValidRandomIdentifier(n.getIdentifier())) {
+            return false;
+        } else {
+            final NodeKey key = new NodeKey(n.getIdentifier());
+            final String source = NodeKey.keyForSourceName(
+                    ((JcrRepository)n.getSession().getRepository())
+                        .getConfiguration()
+                        .getStoreName());
+            return key.getSourceKey() != source;
```
Comment (ajs6f) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/utils/FedoraTypesUtils.java]: Is this `!=` or `!equals()`?

**Comment by ajs6f** [comment]
I love the fact that there are so many more deletions here than additions. That really makes me happy, seriously.

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetBinaryStore.java (hunk 2 above)
Comment (acoburn) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetBinaryStore.java]: Yes it is. http://docs.jboss.org/modeshape/4.2.0.Final/api/org/modeshape/jcr/RepositoryConfiguration.BinaryStorage.html#getBinaryStore()

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/utils/FedoraTypesUtils.java (hunk 5 above)
Comment (acoburn) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/utils/FedoraTypesUtils.java]: My mistake, it should be the latter. More importantly, though, there should be a unit test for this.

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetCacheManager.java (hunk 4 above)
Comment (acoburn) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetCacheManager.java]: This should be IOException

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetBinaryStore.java (hunk 2 above)
Comment (ajs6f) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetBinaryStore.java]: Ooh, grody. Oh well, not our fault.

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetCacheManager.java (hunk 4 above)
Comment (ajs6f) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetCacheManager.java]: Rockin'.

**Comment by acoburn** [comment]
@ajs6f I tightened the implementation, and there are now even more deletions (and fewer additions)

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetCacheManager.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Get the internal Infinispan CacheManager from a Repository
+ *
+ * @author acoburn
+ */
+class GetCacheManager implements Function<Repository, CacheManager> {
+
+    private static final Logger LOGGER =
+            getLogger(GetCacheManager.class);
```
Comment (ajs6f) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/functions/GetCacheManager.java]: I still don't get the line break here. Is your editor set to break at 80 chars or something?

**Comment by ajs6f** [comment]
The code looks rad. But seriously, is your editor config'd funny? All the lines are really short.

**Comment by acoburn** [comment]
@ajs6f lines are stretched out now

**Comment by ajs6f** [comment]
Yay! I'm good with this. I'll let @awoods review and merge, since there is no reviewer on the ticket.

**Comment by awoods** [comment]
Resolved with:
https://github.com/fcrepo4/fcrepo4/commit/e613e6b9907abd8542eb37c08579c4ffbeaeaf18 and
https://github.com/fcrepo4/fcrepo4/commit/fabb0cc01d08c4ffc2dbf3b7e60056810a5f6bc0 and
https://github.com/fcrepo4/fcrepo4/commit/1c0156c1ab8058f9011768ce365a7446b4a496e0

**Pull request closed without merging**